h0rton/h0_inference/h0_posterior.py

```diff
@@ -0,0 +1,269 @@
+        abcd_ordering_i : list
+            ABCD in an increasing dec order if the keys ABCD mapped to values 0123, respectively, e.g. [3, 1, 0, 2] if D (value 3) is lowest, B (value 1) is second lowest
+        
+        """
+        self.z_lens = z_lens
+        self.z_src = z_src
+        self.measured_vd = measured_vd
+        self.measured_vd_err = measured_vd_err
+        self.measured_td = np.array(measured_td)
+        self.measured_td_err = np.array(measured_td_err)
+        self.lens_light_R_sersic = lens_light_R_sersic
+        self.H0_prior = H0_prior
+        self.kappa_ext_prior = kappa_ext_prior
+        self.aniso_param_prior = aniso_param_prior
+        self.kwargs_model = kwargs_model
+        self.abcd_ordering_i = abcd_ordering_i
+
+        self.R_slit = 1.0 # arcsec
+        self.dR_slit = 1.0 # arcsec
+        self.psf_fwhm = 0.6 # arcsec
+        self.Om0 = 0.27 # Omega matter
+        self.Ob0 = 0.0 # Omega baryons
+        # Number of AGN images
+        self.n_img = len(measured_td)
+        # TODO: key checking depending on kwargs_model
+        self.kwargs_lens = [lens_mass_dict, ext_shear_dict]
+        self.set_kwargs_ps(ps_dict)
+        # Always define point source in terms of `LENSED_POSITION` for speed
+        self.kwargs_model.update(dict(point_source_model_list=['LENSED_POSITION']))
+
+        # Pre-store for reordering image arrays
+        dec_image = self.kwargs_ps[0]['dec_image']
+        self.increasing_dec_i = np.argsort(dec_image)
+
+    @classmethod
+    def from_dict(cls, lens_dict):
+        """Initialize H0Posterior from a dictionary
+
+        Parameters
+        ----------
+        lens_dict : dict
+            contains properties required to initialize H0Posterior. See `__init__` method above for the required parameters and their formats.
+
+        """
+        return cls(lens_dict.items())
+
+    def set_kwargs_ps(self, ps_dict):
+        """Sets the kwargs_ps class attribute as those coresponding to the point source model `LENSED_POSITION`
+
+        Parameters
+        ----------
+        ps_dict : dict
+            point source parameters definitions, either of `SOURCE_POSITION` or `LENSED_POSITION`
+
+        """
+        if 'ra_source' in ps_dict:
+            # If the user provided ps_dict in terms of the source position, we precompute the corresponding image positions before we enter the sampling loop.
+            lens_model_class = LensModel(self.kwargs_model['lens_model_list'])
+            ps_class = PointSource(['SOURCE_POSITION'], lens_model_class)
+            kwargs_ps_source = [ps_dict]
+            ra_image, dec_image = ps_class.image_position(kwargs_ps_source, self.kwargs_lens)
+            self.kwargs_ps = [dict(ra_image=ra_image[0],
+                                   dec_image=dec_image[0],
+                                   point_amp=ps_dict['point_amp'])]
```

Comment:
you may not need the 'point_amp' part here but this might be prone to errors as there might be different understanding of what this is when having parameterized the source in the source position (e.g. intrinsic source brightness vs list of magnified image brightnesses)

Reply:
Understood! Deleting as it's not used.